sherpa/sim/__init__.py

```diff
@@ -828,6 +828,12 @@ class ReSampleData(NoNewAttributesAfterInit):
 
     """
     def __init__(self, data, model):
+
+        # Should this error out if data is an instance of Data1DInt?
+        if data.ndim != 1:
+            msg = f"{ReSampleData.__name__} {type(data)}"
```

Comment:
Should we be more wordy here, e.g. "Resample data does not implement support for Data2DInt". I know it's a NotImplementedError, I'm just trying to help a user understand what to do if they hit this error (namely, use a different data class).

Reply:
I don't have a lot of mental energy for improving the user experience here. If this call fails we don't have any replacement code the user could use - they'd have to do it themselves. I'm not sure how best to explain this without writing really-long error messages which then become too wordy. But as I said , I have little energy for this so if you can come up with a sensible message then I'm all ears.

Reply:
```suggestion
            msg = f"{ReSampleData.__name__} is only implemented for 1D data, got {type(data)} instead."
```